File: python/mahasiswa.py
from sivitasakademik import SivitasAkademik
from Courses import Courses


class mahasiswa(SivitasAkademik):

    courses = []
    # constructor mahasiwa

    def __init__(self, nik, name, jenis_kelamin, asal_univ, email, nim, course=Courses):
        # pemanggilan variabel di kelas parent
        super().__init__(nik, name, jenis_kelamin, asal_univ, email)
        self.nim = nim
        self.courses.append(course)

    # disini untuk menambahkan course dimana ini adalah array of object
    def add_course(self, course=Courses):
        self.courses.append(course)
    # getter/setter methods untuk class  mahasiswa

    def get_nim(self):
        return self.nim

    def get_courses(self):
        return self.courses

File: CPP/mahasiswa.cpp
#include "header.hh"
#include <vector>

// constructor for Mahasiswa class
Mahasiswa::Mahasiswa(string nik, string name, string jenis_kelamin, string asal_univ, string email, string nim, Courses course)
    : SivitasAkademik(nik, name, jenis_kelamin, asal_univ, email)
{
    this->nim = nim;
    this->courses.push_back(course);
}

// method to add a course to the Mahasiswa's courses vector
void Mahasiswa::add_course(Courses course)
{
    this->courses.push_back(course);
}

// getter methods for Mahasiswa class
string Mahasiswa::get_nim()
{
    return this->nim;
}

vector<Courses> Mahasiswa::get_courses()
{
    return this->courses;
}